nifi-nar-bundles/nifi-network-bundle/nifi-network-processors/src/main/java/org/apache/nifi/processors/network/ParseNetflowv5.java

```diff
@@ -0,0 +1,258 @@
+                StreamUtils.fillBuffer(in, buffer);
+            }
+        });
+
+        final int processedRecord;
+        try {
+            processedRecord = parser.parse(buffer);
+            getLogger().debug("Parsed {} records from the packet", new Object[] { processedRecord });
+        } catch (Throwable e) {
+            getLogger().error("Parser returned unexpected Exception {} while processing {}; routing to failure", new Object[] { e, flowFile });
+            session.transfer(flowFile, REL_FAILURE);
+            return;
+        }
+
+        try {
+            final List<FlowFile> multipleRecords = new ArrayList<>();
+            switch (destination) {
+            case DESTINATION_ATTRIBUTES:
+                final Map<String, String> attributes = new HashMap<>();
+                generateKV(multipleRecords, session, flowFile, attributes, parser, processedRecord);
+                break;
+            case DESTINATION_CONTENT:
+                generateJSON(multipleRecords, session, flowFile, parser, processedRecord, buffer);
+                break;
+            }
+            // Create a provenance event recording the routing to success
+            multipleRecords.forEach(recordFlowFile -> session.getProvenanceReporter().route(recordFlowFile, REL_SUCCESS));
+            session.getProvenanceReporter().route(flowFile, REL_ORIGINAL);
+            // Ready to transfer and commit
+            session.transfer(flowFile, REL_ORIGINAL);
+            session.transfer(multipleRecords, REL_SUCCESS);
+            session.adjustCounter("Records Processed", processedRecord, false);
+            session.commit();
+        } catch (Exception e) {
+            // The flowfile has failed parsing & validation, routing to failure
+            getLogger().error("Failed to parse {} as a netflowv5 message due to {}; routing to failure", new Object[] { flowFile, e });
+            // Create a provenance event recording the routing to failure
+            session.getProvenanceReporter().route(flowFile, REL_FAILURE);
+            session.transfer(flowFile, REL_FAILURE);
+            session.commit();
+            return;
+        } finally {
+            session.rollback();
+        }
+    }
+
+    private void generateJSON(final List<FlowFile> multipleRecords, final ProcessSession session, final FlowFile flowFile, final Netflowv5Parser parser, final int processedRecord, final byte[] buffer)
+            throws JsonProcessingException {
+        int numberOfRecords = processedRecord;
+        FlowFile recordFlowFile = flowFile;
+        int record = 0;
+        while (numberOfRecords-- > 0) {
+            ObjectNode results = mapper.createObjectNode();
+            // Add Port number and message format
+            results.set("port", mapper.valueToTree(parser.getPortNumber()));
+            results.set("format", mapper.valueToTree("netflowv5"));
+
+            recordFlowFile = session.clone(flowFile);
```

Comment:
I think `session.create(FlowFile)` is the right one to use here because it will establish a parent-child relationship between the newly created record and the input flowfile. I think `clone` also clones the content so that could bog NiFi down a bit as well.

Reply:
I can use session.create(). But do you think any need will arise in future from the user to good to have the attributes from the original flowfile?  If so, we will stick with session.clone() only. whats your view on this?

Reply:
If we need to go there, you can do this later:

```
FlowFile newFF = session.create(inputFlowFile);
newFF = session.putAllAttributes(newFF, inputFlowFile.getAttributes());
```

Reply:
ok cool. let me currently use session.create() alone..